File: stack/stack.c
#include <stdlib.h>
#include "stack.h"


void _add(Stack *head,int value){
    Stack node = (Stack)malloc(sizeof(Stack));
    if(node !=NULL){
      node->value = value;
      node->next = *head;
      *head = node;
    }
}

int _remove(Stack *head){
   int value = -1;
   if(head!=NULL){
        Stack top = *head;
        value = (*head)->value;
        *head = (*head)->next;
        free(top);

    }
    return value;
}

File: stack/stack.js
var StackError = module.exports.StackError = function (message) {
  this.name = 'StackError';
  this.message = message || '';
};

StackError.prototype = new Error();
StackError.prototype.constructor = StackError;

var Stack = module.exports.Stack = function () {
  this.head   = null;
  this.length = 0;
};

Stack.prototype.push = function (value) {
  var node = {
    value: value,
    next: null
  };

  if (this.head === null) {
    this.head = node;
  } else {
    node.next = this.head;
    this.head = node;
  }

  return this.length += 1;
};

Stack.prototype.pop = function () {
  // If there is no head node, return `undefined`
  if (this.head === null) {
    throw new StackError('stack empty on pop');
  }

  var node = this.head;

  // Update the head reference and remove the next node reference from the previous head
  this.head = node.next;
  node.next = null;

  this.length -= 1;

  return node.value;
};
